homeassistant/helpers/trigger.py

```diff
@@ -49,12 +49,29 @@
 )
 
 
+@dataclass
+class Trigger:
```

Comment:
An abstract base class would be better, I think. Then the platform will subclass this class to define the trigger instead of having to create a new module for every trigger.

Reply:
No one is forcing anyone to create new a module per trigger with the current proposal.
But OK, I'll change to a class which contains the trigger implementation instead of a class which just points to functions.